src/map/unit.cpp

```diff
@@ -3412,6 +3428,10 @@ int32 unit_remove_map_(struct block_list *bl, clr_type clrtype, const char* file
 			if (!battle_config.mob_slave_keep_target)
 				md->target_id=0;
 
+			// When a monster is removed from map, its spotted log is cleared
+			for (int32 i = 0; i < DAMAGELOG_SIZE; i++)
+				md->spotted_log[i] = 0;
```

Comment:
On mobile right now, but I think we have that ARRAYCLEAR() macro which could be used here until this gets flipped to a dynamic array type.